File: Sort the Jumbled Numbers.cpp
class Solution {
public:
    int convert(int i, vector<int>& mapping) {
        if (i == 0) return mapping[i];

        int j = 0, m = 0;
        while (i > 0) {
            j += mapping[(i % 10)] * pow(10, m);
            m++;
            i /= 10;
        }
        return j;
    }
    vector<int> sortJumbled(vector<int>& mapping, vector<int>& nums) {
        vector<vector<int>> arr;
        int n = nums.size();
        arr.reserve(n);
        for (int i = 0; i < n; i++) {
            arr.push_back({convert(nums[i], mapping), i, nums[i]});
        }
        sort(arr.begin(), arr.end());

        vector<int> ans;
        ans.reserve(n);
        for (int i = 0; i < n; i++) {
            ans.push_back(arr[i][2]);
        }
        return ans;
    }
};

File: Sort the Jumbled Numbers.go
func powInt(x, y int) int {
    return int(math.Pow(float64(x), float64(y)))
}

func convert(i int, mapping *[]int) int {
    m := *mapping
    if i == 0 {
        return m[i]
    }

    j := 0
    n := 0
    for i > 0 {
        j += m[i % 10] * powInt(10, n)
        n++
        i /= 10
    }
    return j
}

type Val struct {
    C int
    I int
    Num int
}

func sortJumbled(mapping []int, nums []int) []int {
    arr := make([]Val, 0)
    for i, num := range nums {
        arr = append(arr, Val{C: convert(num, &mapping), I:i, Num:num})
    }

    sort.Slice(arr, func (i, j int) bool {
        if arr[i].C == arr[j].C {
            return arr[i].I < arr[j].I
        }
        return arr[i].C < arr[j].C
    })

    ans := make([]int, 0)
    for _, p := range arr {
        ans = append(ans, p.Num)
    }
    return ans
}
